### PR TITLE
Add the SQL storage backend doc

### PR DESCRIPTION
**Pull request by hickeyma**
Add the storage backend section

Merge when:
- [x] #538 is merged
- [x] Helm 3.2 released (PR helm/helm#7635)

### CODE REVIEW

**Comment by hickeyma** [comment]
@emaincourt  Do you have a link to standard documentation for making PostgreSQL DB production ready?

**Comment by emaincourt** [comment]
@hickeyma sorry I did not see your message. I don't think the official PostgreSQL documentation gives much information about it. On my side I would recommend:
- enable SSL
- make regular saves (dumps or snapshots) of the volume(s)
- use connections pooling (e.g. pgBounce)

There are a lot of different options, but they all depend on the use case and the environment. @d4mian anything that you think about ?

**Comment by hickeyma** [comment]
> I don't think the official PostgreSQL documentation gives much information about it. 

@emaincourt Do you mind looking around for some documentation that can be referenced? This would be better the better option if we can.

**Comment by bridgetkromhout** [comment]
@hickeyma do we want to hold these PRs for the additional link you requested? Thanks.

**Comment by hickeyma** [comment]
@bridgetkromhout waiting on feedback from @emaincourt re https://github.com/helm/helm-www/pull/558#issuecomment-617343217 and PR #538

**Comment by hickeyma** [comment]
Ping @emaincourt. Do you mind addressing https://github.com/helm/helm-www/pull/558#issuecomment-617343217? I would like to try and get the SQL storage backend docs merged.

**Comment by emaincourt** [comment]
@hickeyma Sorry for being late, I totally forgot this PR. So as far as I can see there is nothing like an official "How to go to production" documentation. However the following links could be shared.

- [How to properly setup TLS on PostgreSQL](https://www.postgresql.org/docs/12/ssl-tcp.html)
- [How to backup data](https://www.postgresql.org/docs/12/backup.html)
- [How to monitor disk usage](https://www.postgresql.org/docs/12/diskusage.html)
- [How to setup HA, Load Balancing and replication](https://www.postgresql.org/docs/12/high-availability.html)
- [Managing roles](https://www.postgresql.org/docs/12/user-manag.html)
- [Managing databases](https://www.postgresql.org/docs/12/managing-databases.html)

Basically all these topics are being covered in the [Server Administration](https://www.postgresql.org/docs/12/admin.html) part of the documentation.

Do you think this could be enough ?

**Comment by hickeyma** [comment]
@emaincourt Thanks for the details. I think the postgres link will be good.

Do you mind looking at the review comments in #538? This PR depends on it.

**Comment by hickeyma** [comment]
@bridgetkromhout @mattfarina Ready to review when you get a chance.

File: content/en/docs/topics/advanced.md [hunk 1]
```diff
@@ -183,3 +184,38 @@ access to Secret resources. For instance, the default
...
+weighs more than 1MB (in which case, it can't be stored in ConfigMaps/Secrets
+because of internal limits in Kubernetes' underlying etcd key-value store).
+
+To enable the SQL backend, you'll need to deploy a SQL database and set the
+environmental variable `HELM_DRIVER` to `sql`. The DB details are set with the
+environmental variable `HELM_DRIVER_SQL_CONNECTION_STRING`.
+
+You can set it in a shell as follows:
+
+```shell
```
Comment (bridgetkromhout) [inline content/en/docs/topics/advanced.md:203]: We have an ongoing discussion around `console` vs `shell` in https://github.com/helm/helm-www/pull/744.

**Comment by hickeyma** [comment]
Thanks for the review @bridgetkromhout. I have merged as we can tackle the shell/console format once decided.

File: content/en/docs/topics/advanced.md (hunk 1 above)
Comment (hickeyma) [reply content/en/docs/topics/advanced.md:205]: Update once resolved, thanks @bridgetkromhout
